pkg/plugins/pluggable/connection.go

```diff
@@ -286,6 +293,9 @@ func (c *PluginConnection) setupLogCollector(ctx context.Context) {
 // The best way to get that information is to instrument the plugin itself. This is mainly a fallback mechanism to
 // collect logs from an uninstrumented plugin.
 func (c *PluginConnection) collectPluginLogs(ctx context.Context) {
+	c.logsWaitGroup.Add(1)
```

Comment:
What do u think moving this line to line 287? 
That way we know where the goroutine is started